diff --git a/.vscode/launch.json b/.vscode/launch.json
index ce26f83..1e9eae7 100644
--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -7,6 +7,7 @@
       "request": "launch",
       "runtimeExecutable": "${execPath}",
       "args": [
+        "--enable-proposed-api=w00445630.dft-ide",
         "--extensionDevelopmentPath=${workspaceFolder}"
       ],
       "outFiles": [
diff --git a/package.json b/package.json
index 2e2567f..fd5f9a3 100644
--- a/package.json
+++ b/package.json
@@ -23,6 +23,9 @@
   "activationEvents": [
     "onStartupFinished"
   ],
+  "enabledApiProposals": [
+    "terminalDataWriteEvent"
+  ],
   "contributes": {
     "viewsContainers": {
       "activitybar": [
@@ -72,7 +75,7 @@
       "properties": {
         "dftIde.apiBase": {
           "type": "string",
-          "default": "",
+          "default": "http://pandas.hisi.huawei.com/dft-ide-server",
           "description": "DFT IDE backend API base URL, for example http://localhost:8080."
         },
         "dftIde.localProjectsRoot": {
@@ -157,7 +160,7 @@
         },
         "dftIde.gitlabHost": {
           "type": "string",
-          "default": "http://7.227.4.70/test11",
+          "default": "http://code-dg.hisi.huawei.com/DFT_IDE_PROJECTS",
           "description": "gitlab host."
         }
       }
diff --git a/src/extension.ts b/src/extension.ts
index 84e070a..a543731 100644
--- a/src/extension.ts
+++ b/src/extension.ts
@@ -1,6 +1,7 @@
 import * as vscode from 'vscode';
 import * as path from 'path';
 import * as fs from 'fs';
+import * as dotenv from 'dotenv';
 import { submitJob, queryJobStatus, getDonauResources } from './services/donauService';
 import { gitService } from './services/gitService';
 import { obsService } from './services/obsService';
@@ -111,6 +112,16 @@ const pipelineRuntimeService = new PipelineRuntimeService({
 const activeJobTimers = new Map<string, ReturnType<typeof setInterval>>();
 
 export function activate(context: vscode.ExtensionContext) {
+  const isDev = context.extensionMode === vscode.ExtensionMode.Development;
+ 
+  if (isDev) {
+    const envPath = path.join(context.extensionPath, '.env');
+ 
+    if (fs.existsSync(envPath)) {
+      dotenv.config({ path: envPath });
+    }
+  }
+
   context.subscriptions.push(dftDiagnostics);
   context.subscriptions.push(
     vscode.workspace.registerTextDocumentContentProvider(OBS_READONLY_SCHEME, {
diff --git a/src/services/workspaceService.ts b/src/services/workspaceService.ts
index 7e47385..1884ad6 100644
--- a/src/services/workspaceService.ts
+++ b/src/services/workspaceService.ts
@@ -438,16 +438,42 @@ export async function cloneRepoWithTerminal(repoUrl: string, projectPath: string
   });
 
   await new Promise<void>((resolve, reject) => {
-    terminal.sendText(`git clone ${repoUrl}`);
-    const disposable = vscode.window.onDidEndTerminalShellExecution(data => {
-      disposable.dispose();
-      if (data?.exitCode === 0) {
-        terminal.dispose();
-        resolve();
-      } else {
-        reject(new Error('Failed to clone the Git repository. See terminal logs for specific error details'));
-      }
-    });
+    if ('onDidEndTerminalShellExecution' in vscode.window) {
+      const disposable = vscode.window.onDidEndTerminalShellExecution(e => {
+        if (e.terminal === terminal) {
+          disposable.dispose();
+          if (e.exitCode === 0) {
+            terminal.dispose();
+            resolve();
+          } else {
+            reject(new Error('Failed to clone the Git repository. See terminal logs for specific error details'));
+          }
+        }
+      });
+      terminal.sendText(`git clone ${repoUrl}`);
+    } else {
+      let output = '';
+      const disposable = vscode.window.onDidWriteTerminalData(e => {
+        if (e.terminal === terminal) {
+          output += e.data;
+          const lines = output.replace(/(\r\n|\r)/g, '\n').split('\n')
+          const cloneFinishedIndex = lines.indexOf('CLONE_FINISHED');
+          if (cloneFinishedIndex > 0) {
+            disposable.dispose();
+            const result = lines[cloneFinishedIndex - 1].toLowerCase();
+            if (result.includes('receiving objects:') || result.includes('unpacking objects:')) {
+              terminal.dispose();
+              resolve();
+            } else if (result.includes('fatal:')) {
+              reject(new Error(result));
+            } else {
+              reject(new Error('Failed to clone the Git repository. See terminal logs for specific error details'));
+            }
+          }
+        }
+      });
+      terminal.sendText(`git clone ${repoUrl} ; echo "CLONE_FINISHED"`);
+    }
   });
 }
 
@@ -460,7 +486,7 @@ export async function setGitCredentialHelper() {
     }
   } catch {
     try {
-      await executeFileCommand('git', ['config', '--global', 'credential.helper', 'cache']);
+      await executeFileCommand('git', ['config', '--global', 'credential.helper', 'cache --timeout=43200']);
     } catch (error) {
       console.error('Failed to set git credential helper:', error);
     }
